Plugin/CppApi/source/ArcGISArView.cpp

```diff
@@ -321,6 +318,47 @@ DeviceOrientation ArcGISArView::toDeviceOrientation(Qt::ScreenOrientations orien
   }
 }
 
+/*!
+  \internal
+
+  Calculate the origin camera to use in TMCC, by addition of the origin camera (wich contains
+  the initial view of the scene view and calibration) and the location camera (which contains
+  the GPS data).
+  m_originCamera and m_locationCamera can be invalid if not set previously.
+ */
+void ArcGISArView::updateTmccOriginCamera() const
+{
+  if (m_originCamera.isEmpty())
+  {
+    if (m_locationCamera.isEmpty())
+    {
+      // Do nothing.
+    }
+    else
+    {
+      m_tmcc->setOriginCamera(m_locationCamera);
+    }
+  }
+  else
+  {
+    if (m_locationCamera.isEmpty())
+    {
+      m_tmcc->setOriginCamera(m_originCamera);
+    }
+    else
+    {
+      const Point oldLocation = m_locationCamera.location();
+      const Point oldOriginLocation = m_originCamera.location();
+      m_tmcc->setOriginCamera(Camera(oldOriginLocation.y() + oldLocation.y(),
+                                     oldOriginLocation.x() + oldLocation.x(),
+                                     oldOriginLocation.z() + oldLocation.z(),
+                                     m_originCamera.heading() + m_locationCamera.heading(),
+                                     m_originCamera.pitch() + m_locationCamera.pitch(),
+                                     m_originCamera.roll() + m_locationCamera.roll()));
+    }
+  }
+}
```

Comment:
I think the readability of this method is good. You could reduce the lines a bit by returning from the first block, but this suggestion is optional. I think the code is good as it as.

```c++
void ArcGISArView::updateTmccOriginCamera() const
{
  if (m_originCamera.isEmpty())
  {
    if (!m_locationCamera.isEmpty())
    {
      m_tmcc->setOriginCamera(m_locationCamera);
    }
    
    return;
  }

  if (m_locationCamera.isEmpty())
  {
    m_tmcc->setOriginCamera(m_originCamera);
  }
  else
  {
    const Point oldLocation = m_locationCamera.location();
    const Point oldOriginLocation = m_originCamera.location();
    m_tmcc->setOriginCamera(Camera(oldOriginLocation.y() + oldLocation.y(),
                                   oldOriginLocation.x() + oldLocation.x(),
                                   oldOriginLocation.z() + oldLocation.z(),
                                   m_originCamera.heading() + m_locationCamera.heading(),
                                   m_originCamera.pitch() + m_locationCamera.pitch(),
                                   m_originCamera.roll() + m_locationCamera.roll()));
  }
}
```

Reply:
After removing the "// Do nothing." comment, the code is small enough.